src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -390,10 +392,44 @@ public QueryResult query(QueryRequest queryRequest) {
     }
 
     /**
-     * TODO: write Javadoc.
+     * Execute a scan meeting the specs of @{link AmazonDynamoDB}, but scoped to single tenants within a shared table.
+     * If no tenant context is specified, a multi tenant scan is performed over the multi tenant shared table.
+     *
+     * Scans scoped to single tenants is not a performant operation. Multi tenant scans are as bad as scans on any other dynamo
+     * table, ie: not great.
      */
     @Override
     public ScanResult scan(ScanRequest scanRequest) {
+        if (getMtContext().getContextOpt().isEmpty()) {
+            // if we're here, we're doing a multi tenant scan on a shared table
+            Preconditions.checkArgument(mtTables.containsKey(scanRequest.getTableName()), scanRequest.getTableName());
+
+
+            ScanResult scanResult =  getAmazonDynamoDb().scan(scanRequest);
+
+            // given the shared table we're working with,
+            // get the function to map the primary key back to
+            // tuple (tenant, virtual table name, and primary attributes)
+            Function<Map<String, AttributeValue>, FieldValue<?>> fieldMapperFunction =
+                getFieldValueFunction(scanRequest.getTableName());
+            List<Map<String, AttributeValue>> unpackedItems = new ArrayList<>(scanResult.getItems().size());
+            List<String> tenants = new ArrayList<>(scanResult.getItems().size());
+            List<String> virtualTables = new ArrayList<>(scanResult.getItems().size());
+            ScanResult ret = new MtScanResult(scanResult, tenants, virtualTables);
+            for (Map<String, AttributeValue> item : scanResult.getItems()) {
+                // go through each row in the scan, and pull out the tenant and table information from the primary key
+                // to separate attributes in each items map
+                FieldValue virtualFieldKeys = fieldMapperFunction.apply(item);
+                TableMapping tableMappping = getMtContext().withContext(virtualFieldKeys.getContext(),
+                    tableName -> getTableMapping(tableName), virtualFieldKeys.getTableName());
+                Map<String, AttributeValue> unpackedVirtualItem = tableMappping.getItemMapper().reverse(item);
```

Comment:
Are you handling paging(`lastEvaluatedKey`)?

Reply:
so I wanna say that's being delegated to default scan() behavior, so I don't expect that will break. This scan is taking the exact same scan request and result from the lower scanRequest call, and just transforming the item map to use the same keys the tenant created the virtual table metadata with, along side tenant table information in the MtScanResult. Will run into that soon if it does break, when actually trying to use paged scanning to backup a large table.